### PR TITLE
Add push API userVisibleOnly sample

### PR DESCRIPTION
**Pull request by justinlulejian**

### CODE REVIEW

**Comment by justinlulejian** [comment]
@oliverdunk (since I can't seem to add a reviewer)

File: api-samples/push/manifest.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  "name": "Service worker with push notification",
+  "version": "1.0",
+  "manifest_version": 3,
+  "background": {
+    "service_worker": "background.js"
+  },
+  "permissions": [
+    "notifications"
+  ]
+}
```
Comment (oliverdunk) [inline api-samples/push/manifest.json:11]: Nit: Can we add newlines?

File: api-samples/push/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# self.registration.pushManager.subscribe() userVisibleOnly
```
Comment (oliverdunk) [inline api-samples/push/README.md:1]: Can we make this "Service worker with push notification" to be consistent with the manifest?

File: api-samples/push/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+This sample demonstrates using the [Push API](https://developer.mozilla.org/en-US/docs/Web/API/Push_API) in chrome with `self.registration.pushManager.subscribe()` and specifically how to use `userVisibleOnly` to silence required notifications when receiving a push message in a service worker based extension.
```
Comment (oliverdunk) [inline api-samples/push/README.md:3]: I don't think we need this and it allows us to accommodate other Chromium browsers 😄 

```suggestion
This sample demonstrates using the [Push API](https://developer.mozilla.org/en-US/docs/Web/API/Push_API) with `self.registration.pushManager.subscribe()` and specifically how to use `userVisibleOnly` to silence required notifications when receiving a push message in a service worker based extension.
```

File: api-samples/push/README.md (hunk 2 above)
Comment (oliverdunk) [inline api-samples/push/README.md:1]: Could we move this to something like `functional-samples/cookbook.push`? We haven't been perfect but I like trying to keep API sample folder names to be APIs exposed under the `chrome` namespace.

Comment (oliverdunk) [inline api-samples/push/README.md:1]: Could you run the linter with `npm run lint:fix`?

File: api-samples/push/background.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+const applicationServerPublicKey = '<key>';
+
+// Whether the extension will attempt to show a notification or not. Controlled
+// by subscribeUserVisibleOnlyFalse() and subscribeUserVisibleOnlyTrue();
+should_show_notif = true;
+
+// Call this to subscribe the extension to a push notification server with
+// `userVisibleOnly: false` to receive silent messages.
+function subscribeUserVisibleOnlyFalse() {
```
Comment (oliverdunk) [inline api-samples/push/background.js:26]: Given how similar this is to the function below, can we just have one function with a parameter?

File: api-samples/push/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+## Overview
+
+By calling a method in the sample and using an external push server website we can simulate an extension receiving a push message where it is required to emit a notification and where it can bypass that requirement (`userVisibleOnly = false`).
+
+## Running this extension
+
+1. Clone this repository.
+2. Go to [web-push-codelab.glitch.me](web-push-codelab.glitch.me) and copy the “Public Key” to the `applicationServerPublicKey` variable in background.js.
+3. Load this directory in Chrome as an [unpacked extension](https://developer.chrome.com/docs/extensions/mv3/getstarted/development-basics/#load-unpacked).
```
Comment (oliverdunk) [inline api-samples/push/README.md:13]: Could we add a note that on macOS (and maybe other OS') you need to enable notifications for Google Chrome in system preferences? If you don't do that you never see a notification.

**Review by oliverdunk — CHANGES_REQUESTED** [review]
This looks great and it's a really nice idea to use the Glitch demo! Left some suggestions but looking forward to getting this merged.

**Comment by sebastianbenz** [comment]
@patrickkettner FYI

**Comment by justinlulejian** [comment]
@AmySteam Hi Amy! Any comments for this change?

File: functional-samples/cookbook.push/background.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+// https://codelabs.developers.google.com/codelabs/push-notifications
+
+// These methods are meant to be called by the developer following the cookbook.
+/*eslint no-unused-vars: ["error", { "varsIgnorePattern": "subscribeUser" }]*/
+
+const applicationServerPublicKey = '<key>';
+
+// Whether the extension will attempt to show a notification or not. Controlled
+// by subscribeUserVisibleOnlyFalse() and subscribeUserVisibleOnlyTrue(). Not
+// meant to be used in production due to ephemeral nature of service workers.
```
Comment (patrickkettner) [inline functional-samples/cookbook.push/background.js:25]: "Not meant to be used in production due to ephemeral nature of service workers." - why is it in the sample?

File: functional-samples/cookbook.push/background.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+let should_show_notification = true;
+
+function subscribeUser(userVisibleOnly) {
+  should_show_notification = userVisibleOnly;
```
Comment (patrickkettner) [inline functional-samples/cookbook.push/background.js:29]: same comment about change snake case to camel case

File: functional-samples/cookbook.push/background.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+  const applicationServerKey = urlB64ToUint8Array(applicationServerPublicKey);
+  self.registration.pushManager
+    .subscribe({
+      // With our new change, this can be set to false. Before/now it must always
```
Comment (patrickkettner) [inline functional-samples/cookbook.push/background.js:33]: what new change?

File: functional-samples/cookbook.push/background.js [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+      // be set to true otherwise an error will be thrown about permissions denied.
+      userVisibleOnly: userVisibleOnly,
+      applicationServerKey: applicationServerKey
+    })
+    .then(function (subscription) {
```
Comment (patrickkettner) [inline functional-samples/cookbook.push/background.js:38]: Please use async/await

File: functional-samples/cookbook.push/background.js [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+      updateSubscriptionOnServer(subscription);
+    })
+    .catch(function (error) {
+      console.error('[Service Worker] Failed to subscribe, error: ', error);
+    });
+}
+
+function updateSubscriptionOnServer(subscription) {
+  // This data would normally be sent to the server, but for this sample we'll
```
Comment (patrickkettner) [inline functional-samples/cookbook.push/background.js:49]: what data is "this data"?

File: functional-samples/cookbook.push/background.js [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+// Push message event listener.
+self.addEventListener('push', function (event) {
+  console.log('[Service Worker] Push Received.');
+  console.log(
+    `[Service Worker] Push had this data/text: "${event.data.text()}"`
+  );
+
+  const user_visible_notification_title = 'Service Worker Push Sample';
+  const user_visible_notification_options = {
```
Comment (patrickkettner) [inline functional-samples/cookbook.push/background.js:68]: same comment about change snake case to camel case

**Review by patrickkettner — CHANGES_REQUESTED** [review]
Thanks so much for making the demo, @justinlulejian! Few comments, really excited to get this merged in soon :D

File: functional-samples/cookbook.push/background.js (hunk 7 above)
Comment (justinlulejian) [reply functional-samples/cookbook.push/background.js]: PTAL for camel case, I'm a little confused on where you use it. Only for local variables, but not for functions?

File: functional-samples/cookbook.push/background.js (hunk 8 above)
Comment (justinlulejian) [reply functional-samples/cookbook.push/background.js]: Updated comment to refer to change.

File: functional-samples/cookbook.push/background.js (hunk 10 above)
Comment (justinlulejian) [reply functional-samples/cookbook.push/background.js]: Updated PTAL.

File: functional-samples/cookbook.push/background.js (hunk 7 above)
Comment (patrickkettner) [reply functional-samples/cookbook.push/background.js]: Should just always be camel case, I’m not aware of any place that should be using snake case

Comment (justinlulejian) [reply functional-samples/cookbook.push/background.js]: I believe this is addressed then. All variables/functions are camel case now.

**Review by patrickkettner — APPROVED** [review]

**Pull request merged**